api/client/src/main/java/jakarta/websocket/ClientEndpoint.java

```diff
@@ -78,4 +78,16 @@
      *         annotation.
      */
     public Class<? extends ClientEndpointConfig.Configurator> configurator() default ClientEndpointConfig.Configurator.class;
+
+    Extension[] extensions() default {};
+
+    @interface Extension {
```

Comment:
I would consider moving this outside of `ClientEndpoint` so it could be used on the Server side as well, because there is `ServerEndpointConfig.getExtensions` but no equivalent on the `ServerEndpoint` annotation.

Reply:
It is the use case of `ServerEndpointConfig#getExtensions` which does not make sense to me.

But it seems to me that either `ServerEndpointConfig#getExtensions` should be deprecated, or there should be an equivalent API on the `@ServerEndpoint` annotation.

But for example I would imagine something like this, which would allow you to negotiate a specific extension without setting a custom configurator.
```java
@ServerEndpoint(value = "/", extensions = { @Extension(name = "permessage-deflate")})
public class MyServerEndpoint
{
    // ...
}
```

Reply:
I think that would work for `ServerEndpointConfig#getExtensions` (matched with new `@Extension` support in `@ServerEndpoint`) if the intersection of those extensions and what the container supports was passed as `installed` to `ServerEndpointConfig.Configurator.getNegotiatedExtensions`. That would also need explicitly documenting in the spec.

That then begs the question what do we name `@Extension`. We can't use the obvious `jakarta.websocket.Extension` as that clashes with the existing interface. `@ExtensionConfig`?